packages/affiliates/test/mocha/affiliates.js

```diff
@@ -21,129 +32,284 @@ const devRewardsToDistribute = "50000";
+      affiliates = DeployerRewards({
+        queries,
+        empAbi: empAbi,
+        empCreatorAbi: empCreatorAbi,
+        coingecko,
+        synthPrices
+      });
+
+      function makePricesWithValue(count) {
+        return [makePrices(count), affiliates.utils.calculateValue];
+      }
+      function makePrices(count) {
+        return Prices(
+          lodash.times(count, i => {
+            // [timestamp, price]: we ensure price here is not 0 so that calculations come out whole
+            return [i, toWei((i + 1).toString()).toString()];
+          })
+        );
+      }
+      const startTime = 0;
+      const endTime = 10;
+      const empWhitelist = ["a", "b", "c", "d", "e", "f"];
+      balanceHistories = empWhitelist.map(x => [x, EmpBalancesHistory()]);
+      const empDeployers = empWhitelist.map(x => [x, { deployer: x + x }]);
+      totalRewards = "100";
+
+      params = {
+        startTime,
+        endTime,
+        empWhitelist,
+        empCreatorAddress: "creator",
+        snapshotSteps: 1,
+        collateralTokenPrices: empWhitelist.map(() => makePrices(endTime - startTime)),
+        collateralTokenDecimals: empWhitelist.map(() => 18),
+        syntheticTokenPricesWithValueCalculation: empWhitelist.map(() => makePricesWithValue(endTime - startTime)),
+        syntheticTokenDecimals: empWhitelist.map(() => 18),
+        blocks: lodash.times(endTime - startTime, i => ({ timestamp: i, number: i })),
+        balanceHistories,
+        empDeployers,
+        totalRewards
+      };
     });
-  });
-  it("getBalanceHistory", async function() {
-    this.timeout(10000);
-    const result = await affiliates.utils.getBalanceHistory(empContracts[0], startingTimestamp, endingTimestamp);
-    assert.ok(result);
-    assert.ok(result.history.length());
-  });
-  it("getAllBalanceHistory", async function() {
-    this.timeout(10000);
-    const result = await affiliates.utils.getAllBalanceHistories(empContracts, startingTimestamp, endingTimestamp);
-    assert.equal(result.length, empContracts.length);
-    result.forEach(([address, history]) => {
-      assert.ok(address);
-      assert.ok(history);
-      assert.ok(history.history.length());
+    it("should give full rewards to a single emp with balance", function() {
+      // add some balance histories to the emp contracts. this is really what ends up adjusting the distribution.
+      // this adds a single balance history event starting at time 0 for the first emp
+      balanceHistories[0][1].handleEvent(0, {
+        name: "PositionCreated",
+        // creating a position for address "aa" with 2 collateral 1 synthetic
+        args: ["aa", "2", "1"],
+        blockTimestamp: 0
+      });
+      balanceHistories[0][1].finalize();
+      const result = affiliates.utils.calculateRewards(params);
+      assert.equal(result.empPayouts["a"], totalRewards);
+    });
+    it("should split rewards equally between equally funded emps", function() {
```

Comment:
Could be nice to have a test that verified that it's splitting based on synthetics and not collateral -- i.e. make the ratio of synths different from the ratio of collaterals.